src/fluree/db/datatype.cljc

```diff
@@ -132,15 +129,19 @@
         local timezone according to your device."
   [s]
   (when-let [matches (re-matches iso8601-time-re s)]
-    (let [time   (->> matches rest butlast)
-          offset (last matches)
-          [hour min sec] (map #?(:clj  #(Integer/parseInt %)
-                                 :cljs #(left-pad % "0" 2))
-                              time)]
-      #?(:clj  (if offset
-                 (OffsetTime/of hour min sec 0 (ZoneOffset/of ^String offset))
-                 (LocalTime/of hour min sec))
-         :cljs (js/Date. (str "1970-01-01T" hour ":" min ":" sec offset))))))
+    #?(:clj (let [time-parts (->> matches rest butlast)
+                  offset     (last matches)
+
+                  [hours minutes seconds second-fraction]
+                  (->> time-parts
+                       (map #(or % "0"))
+                       (map #(Integer/parseInt %)))
+
+                  nanos (* second-fraction 1000000)]
```

Comment:
This is where I convert decimal seconds into nanoseconds, and I think it's correct, but unit conversion is at the heart of many a weird bug so I'm calling it out.